pkg/server/tls/authority/authority.go

```diff
@@ -242,12 +242,14 @@ func (d *DynamicAuthority) ensureCA(ctx context.Context) error {
 
 	s, err := d.lister.Get(d.SecretName)
 	if apierrors.IsNotFound(err) {
+		d.log.V(logf.DebugLevel).Info("Will regenerate CA", "reason", "CA secret not found")
 		return d.regenerateCA(ctx, nil)
 	}
 	if err != nil {
 		return err
 	}
-	if d.caRequiresRegeneration(s) {
+	if required, reason := caRequiresRegeneration(s); required {
+		d.log.V(logf.DebugLevel).Info("Will regenerate CA", "reason", reason)
```

Comment:
This log level may not be enabled in some installations, while some of the regenerate-reasons are not expected, and are logged at the error level at present. I think it could make sense to return an error to not change existing log behavior too much. WDYT?

Reply:
I reverted the refactor of `caRequiresRegeneration`, we are now not 100% that it returns false for the correct reasons in our unit test, something we can improve later.